server/ai_live_video.go

```diff
@@ -81,14 +82,32 @@ func startTrickleSubscribe(ctx context.Context, url *url.URL, params aiRequestPa
 
 	// read segments from trickle subscription
 	go func() {
+		var err error
 		defer w.Close()
+		defer func() { params.liveParams.errorCallback(err) }() // wrapper function to be able to reference the latest err
+		retries := 0
 		for {
-			segment, err := subscriber.Read()
+			if !params.inputStreamExists() {
+				clog.Errorf(ctx, "Stopping trickle subscribe, input stream does not exist.")
+				break
+			}
+			var segment *http.Response
+			segment, err = subscriber.Read()
 			if err != nil {
+				if errors.Is(err, trickle.EOS) {
+					clog.Infof(ctx, "Trickle subscription end of stream: %s", err)
+					return
+				}
 				// TODO if not EOS then signal a new orchestrator is needed
 				clog.Infof(ctx, "Error reading trickle subscription: %s", err)
-				return
+				if retries > 2 {
+					return
+				}
+				retries++
+				time.Sleep(5 * time.Second)
```

Comment:
Isn't 5s a lot, would it maybe make sense to have it as `1s` and have more retries? Just raising some points, but up to you and @j0sh 🙃

Reply:
Yeah for live video, 5 seconds is a long time ... output GOPs are [currently ~3 seconds](https://github.com/livepeer/ai-worker/blob/bb402bf9b1b55e14b7f619abab804b2ca13b1b41/runner/app/live/trickle/segmenter.py#L19), so we'd be almost 2 segments behind the live edge by the time the sleep completes. I'd do a short sleep, maybe 500ms max, and retry ~5 times